### ISSUE TITLE
Implement player handles

### ISSUE THREAD

**Issue by norswap** [0xFableOrg/0xFable#117]
Two key features:

- resolve players ENS if they have one, use https://viem.sh/docs/ens/actions/getEnsName.html

- Add a `PlayerHandle` contract where users can register a single handle for their address (they should be able to change it as well)
     - Handles should be unique
     - Add a `checkHandleValidity` function that restricts names (e.g. only ASCII, minimum 5 characters, maybe not contain dots (to avoid ENS / other name system clashes)
     - Add a `checkPlayerEligibility` that can be always `true` for now but down the line will only enable players that have completed some milestones (e.g. minimum amount of matches, minimum spend, ...) to register handles to avoid squatting.
     - If a user has an ENS, let them select whether they want to use their handle or their ENS.
     - Add an admin role that can be used later to avoid impersonation and squatting of popular names.

The frontend must be modified to display player handles (or ENS).

**Comment by faytey** [comment]
can i hop on this?

**Comment by norswap** [comment]
Go ham :)

**Comment by faytey** [comment]
Thanks I'd work on this, for better context, I'd be writing a cairo contract to handle Player Ens/profile and updating the frontend showing their ens right?

**Comment by norswap** [comment]
No, we're using Solidity on this game as we target an EVM rollup chain. Cairo contracts only work on Starknet.

**Comment by saimeunt** [comment]
@faytey have you started working on this?
I'd like to take this one if you change your mind and work on something else please.

**Comment by faytey** [comment]
Just seeing this notification @saimeunt yes you can go ahead. @norswap I’d take up other issues thanks

**Comment by faytey** [comment]
If this issue is still up, i would love to continue working on it

**Comment by saimeunt** [comment]
Hey I'm no longer actively working on this feel free to pick it @faytey

**Comment by faytey** [comment]
> Hey I'm no longer actively working on this feel free to pick it @faytey

Cool, @norswap is this okay?

**Comment by norswap** [comment]
Yes, go for it!

**Comment by faytey** [comment]
Hi Sir @norswap I made a PR to this issue here [117](https://github.com/0xFableOrg/0xFable/issues/117) I need your review and comments sir.

**Comment by faytey** [comment]
@norswap 

> Hi Sir @norswap I made a PR to this issue here [117](https://github.com/0xFableOrg/0xFable/issues/117) I need your review and comments sir.

**Comment by norswap** [comment]
Sorry, things have been hectic these days. I will try to review this as soon as possible, but it might still take a few days!
